src/tools/perf/perftest.c

```diff
@@ -424,11 +431,45 @@ static ucs_status_t parse_test_params(ucx_perf_params_t *params, char opt, const
             return -1;
         }
         return UCS_OK;
+    case 'i':
+        params->iov_stride = atol(optarg);
+        if (!params->iov_stride) {
+            return UCS_ERR_INVALID_PARAM;
+        }
+        return UCS_OK;
     case 'n':
         params->max_iter = atol(optarg);
         return UCS_OK;
     case 's':
-        params->message_size = atol(optarg);
+        optarg_ptr = (char *)optarg;
+        token_num  = 0;
+        /* count the number of given message sizes */
+        while((optarg_ptr = strchr(optarg_ptr, delim))) {
```

Comment:
interesting, IMHO this is better:
` while((optarg_ptr = strchr(optarg_ptr, delim)) != NULL) {`